### PR TITLE
[Yaml] Add lint:yaml command usage

### PR DESCRIPTION
**Pull request by chalasr**
This documents https://github.com/symfony/symfony/pull/19139 (merged on 3.2) and closes https://github.com/symfony/symfony/pull/19916.

### CODE REVIEW

File: components/yaml.rst [hunk 1]
```diff
@@ -250,6 +250,59 @@ representation of the object.
     parsers will likely not recognize the ``php/object`` tag and non-PHP
     implementations certainly won't - use with discretion!
 
+Syntax Validation
+~~~~~~~~~~~~~~~~~
+
+The syntax of YAML contents can be validated through the CLI using the
+:class:`Symfony\\Component\\Yaml\Command\\LintCommand` command.
```
Comment (ogizanagi) [inline components/yaml.rst]: Perhaps we should mention the console component is required (not required by the YAML one, except in dev) ?

Comment (chalasr) [inline components/yaml.rst]: Yeah it was expected.. ^^ 
Btw, how would you mention it? I thought about a `caution`

Comment (ogizanagi) [inline components/yaml.rst]: Not sure what is the best way to do it, nor where we can encounter similar cases in the components documentation... Simply adding a little sentence followed by:

```sh
composer require symfony/console
```

?

I quickly searched and found those documentations:

- https://github.com/symfony/symfony-docs/blob/master/console/command_in_controller.rst#showing-colorized-command-output
- https://github.com/symfony/symfony-docs/blob/master/setup/bundles.rst#look-for-deprecations-and-fix-them
- https://github.com/symfony/symfony-docs/blob/master/service_container/lazy_services.rst#installation
- https://github.com/symfony/symfony-docs/blob/master/components/form.rst#csrf-protection

Comment (chalasr) [inline components/yaml.rst]: Perfect, applied. Thank you

**Comment by fabpot** [comment]
👍

**Comment by javiereguiluz** [comment]
I'm not that sure about this. Fabien is right that this is a component and you must integrate it into your setup ... but at the same time, providing a command that you cannot use unless you build a Console application looks not very friendly for the developer.

Could components provide their own commands, ready to use if you also use the Console component?

**Comment by chalasr** [comment]
@javiereguiluz My first thought was the same as yours but indeed, it's tricky. 

Having a `bin/` directory in a component which doesn't contain a `src/` directory feels wrong imho and, even if it's not blocking, introducing this "mistake" just for a `lint` command sounds slightly overkill. 
On the other hand, we just introduced the same `bin/` directory in the phpunit-bridge (which is not a component but has the same structure), but as it contains more logic, it seems to make more sense.

I think providing access to commands in components is already a good thing, people who want to use it just have to copy-paste this snippet. Even if not ideal, that's not a big deal.
I mean: using the components in a standalone way may require some work for reproducing features that are available in the full-stack framework (which has the console as hard dependency and provides a console application).

shall I reopen symfony/symfony/pull/19916 for discussion and deciders opinion?
cc @fabpot

**Comment by fabpot** [comment]
I don't understand the last sentence of @javiereguiluz comments. We do provide the command, and it is ready to use with any console application. The snippet in this PR is just how you can create a standalone script that only validate yaml files. But most of the time, you just need to add this command to your pre-existing console app.

File: components/yaml.rst [hunk 2]
```diff
@@ -250,6 +250,65 @@ representation of the object.
...
+.. code-block:: bash
+
+    $ php lint.php path/to/file.yml --format json
+
+    [
+        {
+            "file": "path/to/file.yml",
+            "valid": true
+        }
+    ]
```
Comment (chalasr) [inline components/yaml.rst]: Do I need something specific for better simulating a console output? 
Maybe just showing the command and removing the previous block showing only the default output could be enough.

File: components/yaml.rst [hunk 3]
```diff
@@ -293,6 +293,53 @@ flag::
...
+The syntax of YAML contents can be validated through the CLI using the
+:class:`Symfony\\Component\\Yaml\\Command\\LintCommand` command.
+
+First, install the Console component:
+
+.. code-block:: bash
+
+    $ composer require symfony/console
+
+Create a script containing a console application:
```
Comment (javiereguiluz) [inline components/yaml.rst]: `Create a script containing a console application:` ->  `Create a console application with lint:yaml as its only command:`  ?

File: components/yaml.rst [hunk 4]
```diff
@@ -293,6 +293,53 @@ flag::
...
+    # validates a single file
+    $ php lint.php path/to/file.yml
+
+    # or all the files in a directory
+    $ php lint.php path/to/directory
+
+    # or contents passed to STDIN
+    $ cat path/to/file.yml | php lint.php
+
+The result will be written on STDOUT, default in TXT format.
```
Comment (javiereguiluz) [inline components/yaml.rst]: Maybe we could replace this:

```
The result will be written on STDOUT, default in TXT format.
For a JSON output, use the ``format`` option:
```

By this:

```
The result is written to STDOUT and uses a plain text format by default.
Add the ``--format`` option to get the output in JSON format:
```

**Review by javiereguiluz — APPROVED** [review]
👍 

Thanks @chalasr.

**Comment by chalasr** [comment]
@javiereguiluz I applied your suggestions, thank you for them.

**Comment by xabbuh** [comment]
Thank you @chalasr.

**Comment by chalasr** [comment]
Looks like gh failed :). Closing as merged.
Thanks for the improvements @javiereguiluz

**Comment by xabbuh** [comment]
@chalasr I didn't merge `3.2` into `master` yet. ;)

**Pull request closed without merging**